src/main/java/HttpServer/RequestParser.java

```diff
@@ -2,26 +2,47 @@
 
 import java.io.IOException;
 import java.util.ArrayList;
+import java.util.HashMap;
 
 public class RequestParser {
```

Comment:
I think it's time to beef up the `RequestParser`, and instead of having it return the response as a `HashMap`, introduce an actual `Request` object, which has methods like, `getUri`, `getMethod`, etc.

Reply:
My initial question is, "How is `response.getUri()` better than `response.get('URI')`?", and my answer to my question is, for the same reason that we use `attr_reader` in Ruby: we can change from getting data to computing data without making changes elsewhere.

Reply:
I dunno though.... I'm still weirded out that we're making an object to map keys to values when we have a data structure that does that already. And my understanding is that headers could be anything, so being able to set arbitrary keys seems appropriate.

Kyle says this is in part to move the data structure into domain language. I'd be interested in hearing more about this suggestion. (I'll go ahead and implement this change though.)

Reply:
`getUri` is better than `get("URI")` for a few reasons. With methods like the former, there is an explicit list of attributes a request has, while the latter puts more onus on clients of `Request`. Also, not all request/response attributes are likely to be strings. e.g. While headers are likely to be strings, the method is likely to be an enum, the status is likely to be an int or an enum, etc.

This is similar to creating a `Board` class in tic tac toe. i.e. The `Board` class is usually just a wrapper for a dictionary.

As far as "moving the data structure into domain language," that is effectively true. Another way of looking at it is the language we use to speak about the domain should be mirrored in the language of the code.